sparkplug-launcher/src/test/scala/springnz/sparkplug/client/ClientExecutorTests.scala

```diff
@@ -0,0 +1,72 @@
+package springnz.sparkplug.client
+
+import org.scalatest._
+import springnz.util.Logging
+
+import scala.concurrent.duration._
+import scala.concurrent.{ Await, Future }
+
+trait ClientExecutableFixture extends BeforeAndAfterEach { this: Suite ⇒
+
+  var executor: ClientExecutor = null
```

Comment:
Global vars and `null` must be banished on principle. A new executor could be created within each test in a Fixture